docs/en/user-manual/deployment_best_practices.md

```diff
@@ -0,0 +1,270 @@
+
+## Security recommendations
+
+Given the nature of the data in the configuration and log files,
+a server should be dedicated for this task and locked down with
+industry best practices. It is recommended that a server that
+sits behind the enterprise firewall be provisioned for this
+application. Only privileged users should be able to connect to
+this machine. A system service account with restricted privileges
+should be created that is specifically intended for running the
+application and writing log files to the system.
+
+The application makes GET and POST requests of the User
+Management API against a HTTPS endpoint. It constructs JSON data
+to represent the changes that need to be written to the Admin
+console, and attaches the data in the body of a POST request to
+the User Management API.
+
+To protect the availability of the Adobe back-end user identity
+systems, the User Management API imposes limits on client access
+to the data.  Limits apply to the number of calls that an
+individual client can make within a time interval, and global
+limits apply to access by all clients within the time period. The
+User Sync tool implements back off and retry logic to prevent the
+script from continuously hitting the User Management API when it
+reaches the rate limit. It is normal to see messages in the
+console indicating that the script has paused for a short amount
+of time before trying to execute again.
+
+Starting in User Sync 2.1, there are two additional techniques available
+for protecting credentials.  The first uses the operating system credential
+store to store individual configuration credential values.  The second uses
+a mechanism you must provide to store the entire configuration file for umapi
+and/or ldap which includes all the credentials required.  These are
+detailed in the next two sections.
+
+### Storing Credentials in OS Level Storage
+
+To setup User Sync to pull credentials from the Python Keyring OS credential store, set the connector-umapi.yml and connector-ldap.yml files as follows:
+
+connector-umapi.yml
+
+	server:
+	
+	enterprise:
+	  org_id: your org id
+	  secure_api_key_key: umapi_api_key
+	  secure_client_secret_key: umapi_client_secret
+	  tech_acct: your tech account@techacct.adobe.com
+	  secure_priv_key_data_key: umapi_private_key_data
+
+Note the change of `api_key`, `client_secret`, and `priv_key_path` to `secure_api_key_key`, `secure_client_secret_key`, and `secure_priv_key_data_key`, respectively.  These alternate configuration values give the key names to be looked up in the user keychain (or the equivalent service on other platforms) to retrieve the actual credential values.  In this example, the credential key names are `umapi_api_key`, `umapi_client_secret`, and `umapi_private_key_data`.
+
+The contents of the private key file is used as the value of `umapi_private_key_data` in the credential store.  This can only be done on platforms other than Windows.  See below for how to secure the
```

Comment:
You might want to just skip describing private key data here and instead always recommend encryption.  And the encryption using PKCS#8 format doesn't have to have a long and complicated passphrase, because it can't be brute forced the way PKCS#5 can.